nixos/modules/services/web-apps/pingvin-share.nix

```diff
@@ -0,0 +1,233 @@
+        example = "medias";
+        description = mdDoc ''
+          Group under which Pingvin Share runs.
+        '';
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = mdDoc ''
+          Whether to open the firewall for the port in {option}`services.pingvin-share.frontend.port`.
+        '';
+      };
+
+      settings = {
+        dataDir = mkOption {
+          type = types.path;
+          default = "/var/lib/pingvin-share";
+          example = "/var/wwww/pingvin";
+          description = mdDoc ''
+            The path to the data directory in which Pingvin Share will store its data.
+          '';
+        };
+
+        hostname = mkOption {
+          type = types.nullOr types.str;
+          default = null;
+          example = "pingvin-share.domain.tdl";
+          description = mdDoc ''
+            The domain name of your instance. If null, the redirections will be made to localhost.
+          '';
+        };
+
+        https = mkOption {
+          type = types.bool;
+          default = false;
+          example = true;
+          description = mdDoc ''
+            Whether to enable HTTPS for the domain.
+          '';
+        };
+      };
+
+      backend = {
+        package = mkPackageOptionMD pkgs [ "pingvin-share" "backend" ] { };
+
+        port = mkOption {
+          type = types.port;
+          default = 8080;
+          example = 9000;
+          description = mdDoc ''
+            The port that the backend service of Pingvin Share will listen to.
+          '';
+        };
+      };
+
+      frontend = {
+        package = mkPackageOptionMD pkgs [ "pingvin-share" "frontend" ] { };
+
+        port = mkOption {
+          type = types.port;
+          default = 3000;
+          example = 8000;
+          description = mdDoc ''
+            The port that the frontend service of Pingvin Share will listen to.
+          '';
+        };
+      };
+
+      nginx = {
+        enable = mkEnableOption (mdDoc ''
+          exposing Pingvin Share with a Nginx reverse proxy.
+        '');
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    users.groups = mkIf (cfg.group == "pingvin") {
+      pingvin = { };
+    };
+
+    users.users = mkIf (cfg.user == "pingvin") {
+      pingvin = {
+        group = cfg.group;
+        # Need to set the user home because the package writes to "$HOME/.npm"
+        home = cfg.settings.dataDir;
+        description = "Pingvin Share daemon user";
+        isSystemUser = true;
+      };
+    };
+
+    systemd.services.pingvin-share-backend = {
+      description = "Backend service of Pingvin Share, a self-hosted file sharing platform and an alternative for WeTransfer.";
+
+      wantedBy = [ "multi-user.target" "pingvin-share-frontend.service" ];
+      wants = [ "network-online.target" ];
+      before = [ "pingvin-share-frontend.service" ];
+      after = [ "network-online.target" ];
```

Comment:
Does the service require to connect to some external service? If not we just need network.target.